pkg/ds/daemon_set.go

```diff
@@ -224,12 +225,24 @@ func (ds *daemonSet) GetNodeSelector() klabels.Selector {
 	return ds.DaemonSet.NodeSelector
 }
 
+func (ds *daemonSet) Manifest() manifest.Manifest {
+	ds.mu.Lock()
+	defer ds.mu.Unlock()
```

Comment:
Why read locks? Is this same mutex used to guard writes too?

Reply:
yeah, https://github.com/square/p2/blob/master/pkg/ds/daemon_set.go#L313

Reply:
although maybe there's only one goroutine acting here?